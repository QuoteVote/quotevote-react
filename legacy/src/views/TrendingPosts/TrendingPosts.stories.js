import React from 'react'

import TrendingPosts from './TrendingPosts'

export default {
  component: TrendingPosts,
  title: 'Trending Posts',
}

export const TrendingPostsComponent = () => <TrendingPosts />
