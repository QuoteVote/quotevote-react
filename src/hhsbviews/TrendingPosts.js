import React, { useState } from 'react'

import {useQuery} from '@apollo/react-hooks';
import Card from "mui-pro/Card/Card.js"
import CardBody from "mui-pro/Card/CardBody.js"
import CustomizedInputBase, { GET_SEARCH_KEY } from 'hhsbComponents/searchBar'
import GridContainer from "mui-pro/Grid/GridContainer.js"
import Pagination from "material-ui-flat-pagination";
import Slider from '@material-ui/core/Slider';

import Calendar from 'hhsbAssets/Calendar.svg'
import Emoji from 'hhsbAssets/FollowingEmoji.svg'
import Filter from 'hhsbAssets/Filter.svg'
import PostsList from 'hhsbComponents/PostsList'

import { GET_TOP_POSTS } from 'graphql/query'

export default function TrendingPosts() {
  const limit = 5
  const [offset, setOffset] = useState(0)
  const [total, setTotal] = useState(1)
  const { data: { searchKey } } = useQuery(GET_SEARCH_KEY)
  const { loading, error, data } = useQuery(GET_TOP_POSTS, {
    variables: { limit, offset: 0, searchKey }
  })
  
  React.useEffect(() => {
    if (data) {
      setTotal(data.total)
    }
  }, [data])

  if (error) return `Something went wrong: ${error}`
  const posts = (data && data.posts) || []

  const handleSlider = (event, newValue) => {
    setOffset(newValue)
  }

  return (
    <Card style={{ display: 'flex', flexBasis: '800px' }}>
      <CardBody>
        <GridContainer
          direction="row"
          justify="center"
          alignItems="center"
        >
          <GridContainer alignItems="center" direction="row" style={{ width: '50%' }}>
            <Slider
              defaultValue={limit}
              value={offset}
              valueLabelDisplay="auto"
              aria-labelledby="range-slider"
              max={total}
              min={1}
              onChange={handleSlider}
            />
          </GridContainer>
          <br></br>
          <br></br>
          <GridContainer
            alignItems="center"
            direction="row"
            justify="space-between"
            style={{
              backgroundColor: '#2A6797',
              boxShadow: '0 6px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)',
              width: '75%',
              wrap: 'nowrap',
            }}
          >
            <h3
              style={{
                color: 'white',
                font: 'League Spartan',
                fontWeight: 'bold',
                paddingLeft: '20px',
                paddingBottom: '5px',
              }}
            >
              Trending Posts
            </h3>

            <div style={{ display: 'flex', justifyContent: 'flex-end', flexDirection: 'row' }}>
              <CustomizedInputBase setOffset={setOffset} />
              <img alt="date" src={Calendar} style={{ display: 'flex', maxHeight: '40px', paddingLeft: '15px' }} />
              <img alt="filter" src={Filter} style={{ display: 'flex', maxHeight: '40px', paddingLeft: '15px' }} />
              <img
                alt="group"
                src={Emoji}
                style={{
                  display: 'flex', maxHeight: '40px', paddingLeft: '15px', paddingRight: '15px',
                }}
              />
            </div>
          </GridContainer>
        </GridContainer>
        <br></br>
        <br></br>
        <PostsList Data={posts} loading={loading} limit={limit} />
      </CardBody>
      <Pagination
        style={{ margin: 'auto' }}
        limit={limit}
        offset={offset}
        total={total}
        onClick={(e, offset) => setOffset(offset)}
      />
    </Card>
  )
}    