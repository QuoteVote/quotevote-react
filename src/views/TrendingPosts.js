import React, { useState } from 'react'

import { useQuery } from '@apollo/react-hooks'
import { useSelector } from 'react-redux'
import Card from 'mui-pro/Card/Card'
import CardBody from 'mui-pro/Card/CardBody'
import { GET_SEARCH_KEY } from 'components/searchBar'
import Pagination from 'material-ui-flat-pagination'
import PostsList from 'components/PostsList'

import { GET_TOP_POSTS } from 'graphql/query'
import SubHeader from 'components/SubHeader'

export default function TrendingPosts() {
  const hiddenPosts = useSelector((state) => state.ui.hiddenPosts)
  const limit = 12 + hiddenPosts.length
  const [offset, setOffset] = useState(0)
  const [total, setTotal] = useState(1)
  const { data: { searchKey } } = useQuery(GET_SEARCH_KEY)
  const { loading, error, data } = useQuery(GET_TOP_POSTS, {
    variables: { limit, offset: 0, searchKey },
  })

  React.useEffect(() => {
    if (data) {
      setTotal(data.total)
    }
  }, [data])

  if (error) return `Something went wrong: ${error}`
  const posts = (data && data.posts) || []

  return (
    <Card style={{ display: 'flex', flexBasis: '800px' }}>
      <CardBody>
        <SubHeader headerName="Trending" />
        <PostsList Data={posts} loading={loading} limit={limit} />
      </CardBody>
      <Pagination
        style={{ margin: 'auto' }}
        limit={limit}
        offset={offset}
        total={total}
        // eslint-disable-next-line
        onClick={(e, offset) => setOffset(offset)}
      />
    </Card>
  )
}
