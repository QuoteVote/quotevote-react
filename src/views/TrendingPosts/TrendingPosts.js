import React, { useState } from 'react'

import { useQuery } from '@apollo/react-hooks'
import moment from 'moment'
import { useSelector } from 'react-redux'
import { GET_SEARCH_KEY } from 'components/SearchBar'
import PostsList from 'components/Post/PostsList'

import { GET_TOP_POSTS } from 'graphql/query'
import SubHeader from 'components/SubHeader'
import { makeStyles } from '@material-ui/core/styles'
import { Grid } from '@material-ui/core'
import FilterInputs from '../../components/Filter/FilterInputs'
import ErrorBoundary from '../../components/ErrorBoundary'

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexGrow: 1,
  },
  list: {
    marginRight: 10,
    maxWidth: '70%',
    [theme.breakpoints.down('sm')]: {
      maxWidth: '100%',
      marginRight: 2,
      marginLeft: 2,
    },
  },
}))
export default function TrendingPosts() {
  const classes = useStyles()
  const hiddenPosts = useSelector((state) => state.ui.hiddenPosts) || []
  const limit = 12 + hiddenPosts.length
  const [offset, setOffset] = useState(0)
  const [dateRangeFilter, setDateRangeFilter] = useState({ startDate: '', endDate: '' })
  const { data: { searchKey } } = useQuery(GET_SEARCH_KEY)
  const variables = {
    limit,
    offset,
    searchKey,
    startDateRange: dateRangeFilter.startDate,
    endDateRange: dateRangeFilter.endDate,
  }

  const {
    loading, error, data, fetchMore,
  } = useQuery(GET_TOP_POSTS, {
    variables,
  })
  const filterState = useSelector((state) => state.filter)

  if (error) return `Something went wrong: ${error}`

  return (
    <ErrorBoundary>
      <Grid
        container
        direction="row"
        justify="center"
        alignItems="center"
        className={classes.root}
        spacing={2}
      >
        <Grid item xs={12}>
          <SubHeader
            headerName="Trending"
            setOffset={setOffset}
            showFilterIconButton={false}
          />
        </Grid>

        {
          filterState.filter.visibility || filterState.date.visibility || filterState.search.visibility ? (
            <Grid item xs={12}>
              <FilterInputs
                filterState={filterState}
                setOffset={setOffset}
                selectAll={null}
                handleSelectAll={() => { }}
                handleActivityEvent={() => { }}
                selectedEvent={null}
                showFilterIconButton={false}
                setDateRangeFilter={setDateRangeFilter}
                dateRangeFilter={dateRangeFilter}
              />
            </Grid>
          ) : null
        }

        <Grid item xs={12} className={classes.list}>
          <PostsList
            data={data}
            loading={loading}
            limit={limit}
            fetchMore={fetchMore}
            variables={variables}
          />
        </Grid>
      </Grid>
    </ErrorBoundary>
  )
}
