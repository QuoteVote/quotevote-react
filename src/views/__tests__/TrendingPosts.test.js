/* eslint-disable no-undef */
/* eslint-disable react/jsx-no-undef */
import React from 'react'
import Card from 'mui-pro/Card/Card'
import TrendingPosts from '../TrendingPosts'

describe('TrendingPosts component unit test', () => {
  let component

  beforeEach(async () => {
    component = mount(
      <ApolloProvider client={client}>
        <Provider store={store}>
          <TrendingPosts />
        </Provider>
      </ApolloProvider>
    )
  })

  it('renders TrendingPosts without crashing', () => {
    expect(component).toMatchSnapshot()
  })

  it('should exist Card tag', async () => {
    const fnd = component.find(Card)
    expect(fnd.exists()).toBeTruthy()
  })
})
