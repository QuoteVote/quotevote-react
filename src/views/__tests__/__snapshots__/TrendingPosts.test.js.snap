// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendingPosts component unit test renders TrendingPosts without crashing 1`] = `ReactWrapper {}`;
